tests/test_api.py

```diff
@@ -55,6 +55,7 @@
   -
 groups:
   -
+default_group: ''
```

Comment:
As noted in the description, I had to add these default_group fields to the tests since the addition of `default_group: str = field(default_factory=str)` into config.py->Config automatically inserts the field into any gitman.yml.

Reply:
That is due to `defaults=True` in the `@datafile` decorator, which was intentional to ensure generated configuration files include all possible options.

Perhaps the attribute could be defined as `default_group: str = 'all'` (but also treat a value of `''` as no default group)?

Reply:
Using '' to indicate no default is current behavior, concerned that 'all' may already be used by some users and will cause aliasing. If we are okay with gitman creating the `default_group : ''` then the following code in config.py already does this: `default_group: str = field(default_factory=str)`.